### PR TITLE
add flake8-alphabetize, significant flake8 fixes

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [x] Tests for the changes have been added (for bug fixes / features)
  - [x] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] CHANGES.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* Adds a flake8 check for alphabetic order of `__all__` and for Exception types
* `xclim.indices.fire._cffwis` had changes in its `__all__` that were slated for a later xclim release; Changes have now been made.
* Significant fixes to the existing code base:
    * PEP8 codes have been cleaned up (obsolete exclusions removed, outstanding issues addressed)
    * `pycodestyle` and `pydocstyle` are now much more explicit in their selections/exclusions
    * Makefile and `tox` linting checks are synchronized
    * Many redundant `noqa` exceptions have been removed from the code base

### Does this PR introduce a breaking change?

Yes. The following hiddent imports were slated for removal from the `__all__` of `xclim.indices.fire._cffwis` and have now been removed:
```python
from xclim.indices.fire._cffwis import (
    _day_length,
    _day_length_factor,
    _drought_code,
    _duff_moisture_code,
    _fine_fuel_moisture_code,
    _overwintering_drought_code,
)
```

### Other information:

We now have a definitive answer on how best to write `*args` and `**kwargs` within a docstring so that it does not raise issues with `pydocstyle` and `sphinx`:
```
r"""Indicator title.

    Parameters
    ----------
    \*\*kwargs
        kwargs description.
```

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -18,15 +18,18 @@ search = __version__ = "{current_version}"
...
+exclude =
 	.git,
 	docs,
 	build,
 	.eggs,
 	tests
 max-line-length = 88
 max-complexity = 12
-ignore = 
+ignore =
```
Comment (bzah) [inline setup.cfg:29]: IMHO it is a good practice to comment which and why rules are ignored.

Comment (Zeitsperre) [reply setup.cfg:29]: Sure thing!

**Review by bzah — APPROVED** [review]
LGTM!

**Review by aulemahal — APPROVED** [review]
Nice!

**Pull request merged**